staging/src/k8s.io/api/discovery/v1/types.go

```diff
@@ -159,10 +159,17 @@ type EndpointConditions struct {
 
 // EndpointHints provides hints describing how an endpoint should be consumed.
 type EndpointHints struct {
-	// forZones indicates the zone(s) this endpoint should be consumed by to
-	// enable topology aware routing.
+	// forZones indicates the zone(s) this endpoint should be consumed by when
+	// using topology aware routing. May contain a maximum of 8 entries.
 	// +listType=atomic
 	ForZones []ForZone `json:"forZones,omitempty" protobuf:"bytes,1,name=forZones"`
+
+	// forNodes indicates the node(s) this endpoint should be consumed by when
+	// using topology aware routing. May contain a maximum of 8 entries.
+	// This is an Alpha feature and is only used when the ImprovedTrafficDistribution
+	// feature gate is enabled.
+	// +listType=atomic
+	ForNodes []string `json:"forNodes,omitempty" protobuf:"bytes,2,name=forNodes"`
```

Comment:
I know you'd have already given this some thought, but wanted us to acknowledge this in the PR too. The question is whether or not to be consistent with the `ForZones` data type. 

The original API design used a dedicated type citing possible future expansion https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/2433-topology-aware-hints#future-api-expansion. 

I can imagine we may never need this for the nodes, but was wondering if it still has some value to be consistent and define a `ForNodes` type?

Reply:
Discussed in the KEP PR [here](https://github.com/kubernetes/enhancements/pull/5140#discussion_r1947996701):

> The `ForZone` struct was apparently added to support future ideas like having multiple zones with `weight` values...
> 
> I guess we could do the same for `ForNodes`... ? I think the idea of weighted hints is based more on the "smart controllers / ambiguous API" model of TopologyAwareHints, while TrafficDistribution has moved more toward "simple controllers / explicit API" so that sort of future API expansion seems unlikely.

OTOH, I did repeatedly find it slightly weird when writing the code that ForNodes didn't behave the same way as ForZones, so maybe I should just make it consistent even if we don't have any use cases for the struct...

Reply:
Thanks for sharing the previous discussion link, sorry missed that conversation.

> OTOH, I did repeatedly find it slightly weird when writing the code that ForNodes didn't behave the same way as ForZones, so maybe I should just make it consistent even if we don't have any use cases for the struct...

Yep, while going through the code is when I too started wondering if having that consistency was a net positive. Seeing you share the same thoughts, I'm in favour of transitioning to a struct.